### PR TITLE
prototype OpenAPI documentation

### PR DESCRIPTION
**Pull request by idoqo**
This is an experiment to explore autogenerating the API docs. Main goals are:
- Autogenerate the documentation.
- Leave as little clutter in the main Go files as possible.

This achieves the two above but has some limitations e.g
- We currently can't specify required vs optional fields in requests (to do that we need to annotate the struct itself).

### CODE REVIEW

File: openapi/docs.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//    SecurityDefinitions:
+//    api_key:
+//      type: apiKey
+//		name: x-api-key
+//		in: header
+//   oauth2:
+//      type: oauth2
+//		authorizationUrl: /auth
+//		tokenUrl: /auth/token
+//		in: header
```
Comment (idoqo) [inline openapi/docs.go:29]: these are just fillers, will figure out how to write them properly later.

File: openapi/docs.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+//     Schemes: http, https
+//     BasePath: /
+//     Version: 0.9.1
+//     Host: localhost:4434
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
```
Comment (viggy28) [inline openapi/docs.go:14]: I remember you mentioned during the last standup that there are a few endpoints that aren't returning `application/json`. Do we need to get that fixed before @idoqo?

File: api/create.go [hunk 3]
```diff
@@ -14,36 +14,32 @@ import (
 	"github.com/spinup-host/spinup/misc"
 )
 
-// Service is used to parse request from JSON payload
+// Cluster is used to parse request from JSON payload
 // todo merge with metastore.ClusterInfo
-type Service struct {
-	UserID   string
+type Cluster struct {
```
Comment (viggy28) [inline api/create.go:19]: +1

File: README.md [hunk 4]
```diff
@@ -106,143 +106,7 @@ postgres=# \dt
 ```
 
 # API Documentation
-## Endpoints
```
Comment (viggy28) [inline README.md:109]: good riddance :)

File: README.md [hunk 5]
```diff
@@ -106,143 +106,7 @@ postgres=# \dt
...
-- Method:
-
-`GET`
-
-- Success Response:
-    - Code: 200
-    - Content: `{jwtofreplaceme}`
+The API documentation is autogenerated and saved in `openapi/swagger.yaml`. To view the documentaion on Swagger UI, run the command below from the project directory:
+```bash
+make serve-docs
```
Comment (viggy28) [inline README.md:111]: serve-docs expects the openapi/swagger.yaml. Should we call `docs` target first when `serve-docs` is invoked?

Also, I had to install `swagger` which I did through `make install-deps`. (Not sure whether we need to mention this)

File: openapi/cluster.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package openapi
+
```
Comment (viggy28) [inline openapi/cluster.go:2]: Is this file auto-generated? If so, would like to know how to generate it :)

**Review by viggy28 — APPROVED** [review]

File: api/create.go [hunk 7]
```diff
@@ -14,36 +14,32 @@ import (
...
-	Memory     int64
-	CPU        int64
-	Monitoring string
+	Memory     int64  `json:"memory,omitempty"`
+	CPU        int64  `json:"cpu,omitempty"`
+	Monitoring string `json:"monitoring"`
 }
 
-
 func (c ClusterHandler) CreateService(w http.ResponseWriter, req *http.Request) {
```
Comment (viggy28) [inline api/create.go:43]: Should we rename this as CreateCluster to stay in sync with the struct name?

File: openapi/docs.go (hunk 2 above)
Comment (idoqo) [reply openapi/docs.go:14]: yeah right, creating an issue for that now.

File: README.md (hunk 5 above)
Comment (idoqo) [reply README.md:111]: not really, but we can mention that. The docs is added to git so `make serve-docs` will work. `make docs` is needed when the documentation needs to be regenerated.

File: openapi/cluster.go (hunk 6 above)
Comment (idoqo) [reply openapi/cluster.go:2]: No sadly :(, the comments are go-swagger annotations (https://goswagger.io/use/spec.html) so I'm currently trying to get used to it too.

File: README.md (hunk 5 above)
Comment (viggy28) [reply README.md:111]: makes sense.

**Review by skant7 — APPROVED** [review]

**Pull request merged**